src/Microsoft.AspNetCore.Testing/Tracing/EventSourceTestBase.cs

```diff
@@ -0,0 +1,37 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Tracing;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Testing.Tracing
+{
+    // This collection attribute is what makes the "magic" happen. It forces xunit to run all tests that inherit from this
+    // base class sequentially, preventing conflicts (since EventSource/EventListener is a process-global concept).
+    [Collection("Microsoft.AspNetCore.Testing.Tracing.EventSourceTests")]
```

Comment:
Now when I think more about it, even if event source tests are ran sequentially enabled event sources might cause other tests to log more events

Reply:
Event  source tests have to run sequentially and no other tests should be running at the same time.

Reply:
True... maybe the async local is the only reliable way to isolate them then.

Reply:
I still like the base class patterns I got in the second iteration though, so I'll preserve that.